### DISCUSSION TITLE
won't lazy evaluation lead to FOUC ?

### DISCUSSION THREAD

**Discussion by lili21** [cssinjs/jss#1490] [Q&A]
Lazy evaluation - Style Sheets get created when a component gets mounted.

**Comment by kof** [comment]
It happens inside useLayoutEffect when used hooks or componentDidMount if used HOC interface as of today. Both are called by react synchronously after inserting the DOM node, so browser has not yet calculated or drawn the styles when we insert them. This is why there should be no FOUC.
